Source/panels/charpanel.cpp

```diff
@@ -0,0 +1,315 @@
+#include "panels/charpanel.hpp"
+
+#include <fmt/format.h>
+#include <string>
+
+#include "DiabloUI/art.h"
+#include "DiabloUI/art_draw.h"
+
+#include "control.h"
+#include "engine/render/cel_render.hpp"
+#include "engine/render/text_render.hpp"
+#include "utils/language.h"
+
+namespace devilution {
+
+struct colorAndText {
+	UiFlags color;
+	std::string text;
+};
+
+struct panelEntry {
+	std::string label;
+	Displacement position;
+	int length;
+	Displacement labelOffset; // label's offset (end of the label vs the beginning of the stat box)
+	int labelLength;          // max label's length - used for line wrapping
+	int labelSpacing;
+	int statSpacing;
+	bool centered;
+	// must be set to true for stat boxes or they don't line up with the "spend stat" button
+	bool high;                                     // if the box should be using the 27px version or 26px
+	std::function<colorAndText()> statDisplayFunc; // function responsible for displaying stat
+};
+
+auto &myPlayer = Players[MyPlayerId];

```

Comment:
This won't work in multiplayer as you get assigned a different id when you aren't the host. Also if you make this a global then put it next to `MyPlayerId` as it is use full for all the other places where it is being used:

```suggestion
PlayerStruct *MyPlayer = &Players[MyPlayerId];
```